### PR TITLE
Issue 1904: Discription for Subcommands

### PR DESCRIPTION
**Pull request by tinkoosingh**

### CODE REVIEW

File: cli/hzn.go [hunk 1]
```diff
@@ -3,11 +3,14 @@ package main
 
 import (
 	"flag"
-	"github.com/open-horizon/anax/cli/sdo"
-	"github.com/open-horizon/anax/version"
 	"os"
 	"strings"
 
+	"github.com/open-horizon/anax/cli/sdo"
```
Comment (dabooz) [inline cli/hzn.go:9]: Did you move these lines around or did the formatting tool do it?

File: cli/hzn.go [hunk 2]
```diff
@@ -70,6 +72,32 @@ func main() {
 	// Command flags and args - see https://github.com/alecthomas/kingpin
 	app := kingpin.New("hzn", msgPrinter.Sprintf(`Command line interface for Horizon agent. Most of the sub-commands use the Horizon Agent API at the default location http://localhost (see environment Environment Variables section to override this).
 
+Subcommands Description:
+  evn: Show the Horizon Environment Variables.
```
Comment (dabooz) [inline cli/hzn.go:76]: typo: evn -> env

File: cli/hzn.go [hunk 3]
```diff
@@ -70,6 +72,32 @@ func main() {
...
+  version: Show the Horizon version.
+  architecture: Show the architecture of this machine ( as defined by Horizon and golang).
```
Comment (dabooz) [inline cli/hzn.go:78]: typo: "( as defined" -> "(as defined"

File: cli/hzn.go [hunk 4]
```diff
@@ -70,6 +72,32 @@ func main() {
...
+  exchange: List and manage Horizon Exchange resources.
+  reginput: Create an input file template for this pattern that can be used for the 'hzn register' command (once filled in). 
+            This examines the services that the specified pattern uses, and determines the node owner input that is required for them.
+  key: List and manage keys for signing and verifying services.
+  register: Register this edge node with horizon.
```
Comment (dabooz) [inline cli/hzn.go:83]: Register this edge node with the management hub.

File: cli/hzn.go [hunk 5]
```diff
@@ -70,6 +72,32 @@ func main() {
...
+  userinput: List or manager the service user inputs that are currently registered on this Horizon edge node.
+  service: List or manage the services that are currently registered on this Horizon edge node.
+  unregister: Unregister and reset this Horizon edge node so that it is ready to be registered again.
+  status: Display the current horizon internal status for the node.
+  eventlog: List the event logs for the current or all registrations.
+  dev: Deployment tools for creation of services.
+  agbot: List and manage Horizon agreement bot resources.
+  util: Utility commands.
+  mms: List and manage Horizon Model Management Service resources.
+  voucher:	List and manage Horizon SDO ownership vouchers.
```
Comment (dabooz) [inline cli/hzn.go:99]: Please remove extra spaces after voucher:

File: cli/hzn.go [hunk 6]
```diff
@@ -70,6 +72,32 @@ func main() {
 	// Command flags and args - see https://github.com/alecthomas/kingpin
 	app := kingpin.New("hzn", msgPrinter.Sprintf(`Command line interface for Horizon agent. Most of the sub-commands use the Horizon Agent API at the default location http://localhost (see environment Environment Variables section to override this).
 
+Subcommands Description:
```
Comment (dabooz) [inline cli/hzn.go:75]: Can you find a way to display this list and the list in the "Commands:" section in alphabetic order?

**Review by dabooz — CHANGES_REQUESTED** [review]

File: cli/hzn.go (hunk 1 above)
Comment (tinkoosingh) [reply cli/hzn.go:9]: it is done by formatting tool

File: cli/hzn.go (hunk 6 above)
Comment (tinkoosingh) [reply cli/hzn.go:75]: yes I can do that @dabooz

File: cli/hzn.go [hunk 7]
```diff
@@ -70,6 +69,32 @@ func main() {
...
+  agbot: List and manage Horizon agreement bot resources.
+  agreement: List or manage the active or archived agreements this edge node has made with a Horizon agreement bot.
+  architecture: Show the architecture of this machine (as defined by Horizon and golang). 
+  attribute: List or manage the global attributes that are currently registered on this Horizon edge node.
+  deploycheck: Check deployment compatibility.
+  dev: Deployment tools for creation of services.
```
Comment (dabooz) [inline cli/hzn.go:78]: Development tools for creation of services.

**Review by dabooz — APPROVED** [review]

**Pull request merged**